be/src/vec/exec/scan/new_jdbc_scanner.cpp

```diff
@@ -0,0 +1,148 @@
+    if (_tuple_desc == nullptr) {
+        return Status::InternalError("Failed to get tuple descriptor.");
+    }
+
+    // get jdbc table info
+    const JdbcTableDescriptor* jdbc_table =
+            static_cast<const JdbcTableDescriptor*>(_tuple_desc->table_desc());
+    if (jdbc_table == nullptr) {
+        return Status::InternalError("jdbc table pointer is NULL of VJdbcScanNode::prepare.");
+    }
+    _jdbc_param.driver_class = jdbc_table->jdbc_driver_class();
+    _jdbc_param.driver_path = jdbc_table->jdbc_driver_url();
+    _jdbc_param.resource_name = jdbc_table->jdbc_resource_name();
+    _jdbc_param.driver_checksum = jdbc_table->jdbc_driver_checksum();
+    _jdbc_param.jdbc_url = jdbc_table->jdbc_url();
+    _jdbc_param.user = jdbc_table->jdbc_user();
+    _jdbc_param.passwd = jdbc_table->jdbc_passwd();
+    _jdbc_param.tuple_desc = _tuple_desc;
+    _jdbc_param.query_string = std::move(_query_string);
+
+    _jdbc_connector.reset(new (std::nothrow) JdbcConnector(_jdbc_param));
+    if (_jdbc_connector == nullptr) {
+        return Status::InternalError("new a jdbc scanner failed.");
+    }
+
+    _is_init = true;
+    return Status::OK();
+}
+
+Status NewJdbcScanner::open(RuntimeState* state) {
+    VLOG_CRITICAL << "NewJdbcScanner::open";
+    if (state == nullptr) {
+        return Status::InternalError("input pointer is NULL of VJdbcScanNode::open.");
+    }
+
+    if (!_is_init) {
+        return Status::InternalError("used before initialize of VJdbcScanNode::open.");
+    }
+    RETURN_IF_CANCELLED(state);
+    RETURN_IF_ERROR(VScanner::open(state));
+    SCOPED_CONSUME_MEM_TRACKER(_mem_tracker);
+    RETURN_IF_ERROR(_jdbc_connector->open());
+    RETURN_IF_ERROR(_jdbc_connector->query());
+    return Status::OK();
+}
+
+Status NewJdbcScanner::_get_block_impl(RuntimeState* state, Block* block, bool* eof) {
+    VLOG_CRITICAL << "NewJdbcScanner::_get_block_impl";
+    if (nullptr == state || nullptr == block || nullptr == eof) {
+        return Status::InternalError("input is NULL pointer");
+    }
+
+    if (!_is_init) {
+        return Status::InternalError("used before initialize of VJdbcScanNode::get_next.");
+    }
+    auto column_size = _tuple_desc->slots().size();
+    std::vector<MutableColumnPtr> columns(column_size);
+    bool mem_reuse = block->mem_reuse();
+    // only empty block should be here
+    DCHECK(block->rows() == 0);
+
+    bool jdbc_eos = false;
+    do {
+        RETURN_IF_CANCELLED(state);
+
+        columns.resize(column_size);
+        for (auto i = 0; i < column_size; i++) {
+            if (mem_reuse) {
+                columns[i] = std::move(*block->get_by_position(i).column).mutate();
+            } else {
+                columns[i] = _tuple_desc->slots()[i]->get_empty_mutable_column();
+            }
+        }
+
+        RETURN_IF_ERROR(_jdbc_connector->get_next(&jdbc_eos, columns, state->batch_size()));
+
+        if (jdbc_eos) {
+            *eof = true;
```

Comment:
set eof to true iff `jdbc_eos` is true and block is empty.
Same as I comment in odbc scanner #12899